test/functional/tests/misc/test_device_capabilities.py

```diff
@@ -0,0 +1,229 @@
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(core_device, cache_device, cache, core, error_output)
+        os_utils.unload_kernel_module("scsi_debug")
+
+    core_device = TestRun.disks['core']
+    max_io_size_path = os.path.join(disk_utils.get_sysfs_path(core_device.get_device_id()),
+                                    'queue/max_sectors_kb')
+    max_io_size = fs_utils.read_file(max_io_size_path)
+
+    with TestRun.group(f"Standard core device | max_sectors_kb default ({max_io_size}KiB)"):
+        with TestRun.step("Start cache and add prepared core device as core."):
+            cache, core, error_output = prepare_cas_device(cache_device, core_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(cache_device, core_device, cache, core, error_output)
+        with TestRun.step("Recreate CAS device with switched cache and core devices."):
+            cache, core, error_output = prepare_cas_device(core_device, cache_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(core_device, cache_device, cache, core, error_output)
+
+    with TestRun.group(f"Standard core device | max_sectors_kb = 128"):
+        core_device.set_max_io_size(Size(128, Unit.KibiByte))
+        with TestRun.step("Start cache and add prepared core device as core."):
+            cache, core, error_output = prepare_cas_device(cache_device, core_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(cache_device, core_device, cache, core, error_output)
+        with TestRun.step("Recreate CAS device with switched cache and core devices."):
+            cache, core, error_output = prepare_cas_device(core_device, cache_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(core_device, cache_device, cache, core, error_output)
+
+
+# Methods used in test
+
+def create_scsi_debug_device(sector_size: int, physblk_exp: int, dev_size_mb=1024):
+    scsi_debug_params = {
+        "delay": "0",
+        "virtual_gb": "200",
+        "dev_size_mb": str(dev_size_mb),
+        "sector_size": str(sector_size),
+        "physblk_exp": str(physblk_exp)
+    }
+    scsi_debug = TestRun.plugin_manager.get_plugin('scsi_debug')
+    scsi_debug.params = scsi_debug_params
+    scsi_debug.post_setup()
```

Comment:
Running plugin `post_setup()` explicitly from the test doesn't seem right. If we need module reload functionality provided by plugin, then plugin should have dedicated method for that. It should not be done through hack on core plugin API.